src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,206 @@
+import { useEffect, useState } from 'react';
+
+import { useParams, useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
 
+import { Person } from '../types';
+import { getPeople } from '../api';
+
+type PageState =
+  | 'loading'
+  | 'error'
+  | 'nothingFound'
+  | 'changeFilter'
+  | 'showTable';
+
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>();
+  const [filteredList, setFilteredList] = useState<Person[]>();
+  const [pageState, setPageState] = useState<PageState>('loading');
+  const [urlSlug, setUrlSlug] = useState<string>();
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const { slug } = useParams();
+
+  // Here it'll take the response from API and add father and mother to
+  // each person, if they exist...
+  const addMotherAndFather = (response: Person[]): Person[] => {
+    const peopleList = response.map((person) => {
+      const findMother = person.motherName
+        ? response.filter((p) => p.name === person.motherName)[0]
+        : null;
+
+      const findFather = person.fatherName
+        ? response.filter((p) => p.name === person.fatherName)[0]
+        : null;
+
+      return {
+        ...person,
+        ...(findFather && { father: findFather }),
+        ...(findMother && { mother: findMother }),
```

Comment:
guess it's ok, yet you could normalise the whole dataset so that props `father` & `mother` are on all records, just some of those will be `null`
```suggestion
        ...{ father: findFather },
        ...{ mother: findMother },
```